subsys/net/lib/dhcpv4/dhcpv4_server.c

```diff
@@ -467,9 +484,11 @@ static uint8_t *dhcpv4_encode_requested_params(
 		case DHCPV4_OPTIONS_SUBNET_MASK:
 			buf = dhcpv4_encode_subnet_mask_option(
 				buf, buflen, &ctx->netmask);
-			if (buf == NULL) {
-				goto out;
-			}
```

Comment:
Please don't remove this check, we need to break the loop in case option encoding fails.